src/dex/dexalot/dexalot.ts

```diff
@@ -0,0 +1,935 @@
+          pairsIntervalMs: DEXALOT_API_PAIRS_POLLING_INTERVAL_MS,
+          pricesIntervalMs: DEXALOT_API_PRICES_POLLING_INTERVAL_MS,
+          blacklistIntervalMs: DEXALOT_API_BLACKLIST_POLLING_INTERVAL_MS,
+          pairsReqParams: this.getAPIReqParams('api/rfq/pairs', 'GET'),
+          pricesReqParams: this.getAPIReqParams('api/rfq/prices', 'GET'),
+          blacklistReqParams: this.getAPIReqParams('api/rfq/blacklist', 'GET'),
+          pairsCacheKey: this.pairsCacheKey,
+          pairsCacheTTLSecs: DEXALOT_PAIRS_CACHES_TTL_S,
+          pricesCacheKey: this.pricesCacheKey,
+          pricesCacheTTLSecs: DEXALOT_PRICES_CACHES_TTL_S,
+          tokensAddrCacheKey: this.tokensAddrCacheKey,
+          tokensCacheKey: this.tokensCacheKey,
+          tokensCacheTTLSecs: DEXALOT_TOKENS_CACHES_TTL_S,
+          blacklistCacheKey: this.blacklistCacheKey,
+          blacklistCacheTTLSecs: DEXALOT_BLACKLIST_CACHES_TTL_S,
+        },
+      },
+    );
+  }
+
+  async initializePricing(blockNumber: number): Promise<void> {
+    if (!this.dexHelper.config.isSlave) {
+      this.rateFetcher.start();
+    }
+
+    return;
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  getPairString(baseToken: Token, quoteToken: Token): string {
+    return `${baseToken.symbol}/${quoteToken.symbol}`.toLowerCase();
+  }
+
+  async getPairData(
+    srcToken: Token,
+    destToken: Token,
+  ): Promise<PairData | null> {
+    const normalizedSrcToken = this.normalizeToken(srcToken);
+    const normalizedDestToken = this.normalizeToken(destToken);
+    if (normalizedSrcToken.address === normalizedDestToken.address) {
+      return null;
+    }
+
+    const cachedTokens = (await this.getCachedTokens()) || {};
+    if (
+      !(normalizedSrcToken.address in cachedTokens) ||
+      !(normalizedDestToken.address in cachedTokens)
+    ) {
+      return null;
+    }
+    normalizedSrcToken.symbol = cachedTokens[normalizedSrcToken.address].symbol;
+    normalizedDestToken.symbol =
+      cachedTokens[normalizedDestToken.address].symbol;
+
+    const cachedPairs = (await this.getCachedPairs()) || {};
+
+    const potentialPairs = [
+      {
+        identifier: this.getPairString(normalizedSrcToken, normalizedDestToken),
+        isSrcBase: true,
+      },
+      {
+        identifier: this.getPairString(normalizedDestToken, normalizedSrcToken),
+        isSrcBase: false,
+      },
+    ];
+
+    for (const pair of potentialPairs) {
+      if (pair.identifier in cachedPairs) {
+        const pairData = cachedPairs[pair.identifier];
+        pairData.isSrcBase = pair.isSrcBase;
+        return pairData;
+      }
+    }
+    return null;
+  }
+
+  getIdentifier(baseSymbol: string, quoteSymbol: string) {
+    // return `${this.dexKey}_${baseSymbol}_${quoteSymbol}`.toLowerCase();
+    // To disable Dexalot in multi/mega routes
+    return this.dexKey;
+  }
```

Comment:
I am not sure this solution works fully. Yeah, it works in default case, but if `includeContractMethods=megaSwap` is provided, I still see Dexalot

Reply:
not actual anymore